### PR TITLE
Add Wishart distribution and tests

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Needs https://github.com/gonum/matrix/pull/414 and https://github.com/gonum/matrix/pull/415

File: distmat/wishart.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	nu  float64
+	src *rand.Rand
+
+	dim     int
+	cholv   mat64.Cholesky
+	logdetv float64
+	upper   mat64.TriDense
+
+	v    *mat64.SymDense // only stored if needed
+	once sync.Once
```
Comment (kortschak) [inline distmat/wishart.go]: I think normally guards like mutexes and onces sit above the guarded fields.

Comment (btracey) [inline distmat/wishart.go]: Done.

File: distmat/wishart.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+// Wishart is a distribution over d×d positive symmetric definite matrices. It
+// is parametrized by a scalar degrees of freedom parameter ν and a d×d positive
+// definite matrix V.
+//
+// The Wishart PDF is given by
+//  p(X) = [|X|^((ν-d-1)/2) * exp(-tr(V^-1 * X)/2)] / [2^(n*d/2) * |V|^(n/2) * Γ(d, ν/2)]
+// where tr is the trace and Γ_p is the multivariate gamma function.
+//
+// nu must be greater than p-1.
```
Comment (sbinet) [inline distmat/wishart.go]: perhaps add a link to wikipedia?

https://en.wikipedia.org/wiki/Wishart_distribution

File: distmat/wishart.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+		dim:     dim,
+		cholv:   chol,
+		logdetv: chol.LogDet(),
+		upper:   u,
+	}
+	return w, true
+}
+
+// setSigma computes and stores the covariance matrix of the distribution.
```
Comment (sbinet) [inline distmat/wishart.go]: `s/setSigma/setV/`

File: distmat/wishart.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	"github.com/gonum/stat/distuv"
+)
+
+// Wishart is a distribution over d×d positive symmetric definite matrices. It
+// is parametrized by a scalar degrees of freedom parameter ν and a d×d positive
+// definite matrix V.
+//
+// The Wishart PDF is given by
+//  p(X) = [|X|^((ν-d-1)/2) * exp(-tr(V^-1 * X)/2)] / [2^(n*d/2) * |V|^(n/2) * Γ(d, ν/2)]
+// where tr is the trace and Γ_p is the multivariate gamma function.
```
Comment (vladimir-ch) [inline distmat/wishart.go]: - and what is p (apart from the notation for PDF)?

Comment (btracey) [inline distmat/wishart.go]: I'll add it to the comment, but X is a positive definite matrix of the size of V

I'll fix Gamma_p. Wikipedia writes it that way to disambiguate from the non-multivariate Gamma function.

Comment (btracey) [inline distmat/wishart.go]: Done on all accounts.

File: distmat/wishart.go (hunk 2 above)
Comment (btracey) [inline distmat/wishart.go]: Done.

Comment (btracey) [inline distmat/wishart.go]: Done.

File: distmat/wishart.go (hunk 3 above)
Comment (btracey) [inline distmat/wishart.go]: Done.

**Comment by sbinet** [comment]
LGTM

**Pull request merged**